File: app/ui/ui.py
import sys

from PyQt6.QtWidgets import QApplication, QStyleFactory
from PyQt6.QtGui import QFontDatabase

from pathlib import Path

from .pages.main_page import MainWindow


def show_ui() -> None:
    app = QApplication(sys.argv)
    app.setStyle(QStyleFactory.create("WindowsVista"))

    inter_id = QFontDatabase.addApplicationFont(((
        Path(__file__)
        .parent
        .parent
        / "fonts"
        / "inter.ttf"
    ).resolve().__str__()))

    if inter_id != -1:
        inter_family = QFontDatabase.applicationFontFamilies(inter_id)[0]
        font_stack = f"'{inter_family}', system-ui"
    else:
        font_stack = "system-ui"

    main_win = MainWindow(font_stack)
    main_win.show()

    exit_code = app.exec()
    sys.exit(exit_code)

File: src/ui/ui.cpp
#include <ui/ui.h>
#include <ui/mainPage.h>

#include <QApplication>
#include <QStyleFactory>
#include <QFontDatabase>
#include <QFont>

int show_ui() {
    int argc = 0;
    char** argv = nullptr;
    QApplication app(argc, argv);
    app.setStyle(QStyleFactory::create("WindowsVista"));

    int fontId = QFontDatabase::addApplicationFont("./src/fonts/inter.ttf");
    if (fontId != -1) {
        QStringList fontFamilies = QFontDatabase::applicationFontFamilies(fontId);
        if (!fontFamilies.isEmpty()) {
            QString fontFamily = fontFamilies.at(0);
            QFont interFont(fontFamily);
            app.setFont(interFont);
        }
    }

    MainWindow main_win;
    main_win.show();

    return app.exec();
}
